src/rust/src/lazy/dsl.rs

```diff
@@ -1600,8 +1600,7 @@ impl Expr {
         self.0.clone().is_finite().into()
     }
 
-    // TODO: rename to is_first_distinct
-    pub fn is_first(&self) -> Self {
+    pub fn is_first_distinct(&self) -> Self {
```

Comment:
can you return `RResult<Self>`?

Reply:
There are a large number of other functions that return Self in the same way, so if we want to do that, it would be better to do that in a separate PR.

Reply:
I think just `Self` is correct where there is no fallibility due to `robj_to` conversion or other error sources. Then the function does not need to wrap return type in a `RResult`.

Reply:
Ah right